deploy/examples/create-external-cluster-resources.py

```diff
@@ -1451,6 +1472,25 @@ def _gen_output_map(self):
                 self.out_map["CSI_CEPHFS_PROVISIONER_SECRET"],
                 self.out_map["CSI_CEPHFS_PROVISIONER_SECRET_NAME"],
             ) = self.create_cephCSIKeyring_user("client.csi-cephfs-provisioner")
+            # create the default "csi" subvolumegroup
+            self.get_or_create_subvolume_group(
+                "csi", self._arg_parser.cephfs_filesystem_name
+            )
+            # pin the default "csi" subvolumegroup
+            self.pin_subvolume(
+                "csi", self._arg_parser.cephfs_filesystem_name, "distributed", "1"
+            )
```

Comment:
Is this run every reconciliation loop?

I'm confused by directory layout where this file is located. I would assume `deploy/examples` is ... example code or something.

Reply:
> Is this run every reconciliation loop?
> 
> I'm confused by directory layout where this file is located. I would assume `deploy/examples` is ... example code or something.

This is meant only for external cephcluster.
This is a script run once during creation to create resources/import of details or upgrade between major versions to update/create newer required resources from external cephcluster.
(Existing external deployments requires the storage(ceph) admin to manually execute the required cmd)

For rook deployed and managed cephclusters, I've opened a feature request here https://github.com/rook/rook/issues/12607.